### PR TITLE
Code simplify

### PR DESCRIPTION
**Pull request by v-shikhanov**
Muti-threading is updated to improve stability.

I've updated interface with new searching features and implemented it.

Program working good and stable.
But I have a little problem: In UI I have three basic elements: menu bar, buttons bar, textarea.
I wanted to create own classes for menu bar and buttons bar, to improve code readability, but I have a problem with dispose method, wich invoked from action of exit button in menu. Dispose method must be called from class instance and can't being called from static content. How could I solve it?

Best regards,
Thanks for previous review.

### CODE REVIEW

File: src/editor/files/FileChooser.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        } else {
+            fileChooser = new JFileChooser((FileSystemView.getFileSystemView().getHomeDirectory()));
+        }
+
+        fileChooser.setDialogTitle("Save text file");
+        fileChooser.setDialogType(JFileChooser.SAVE_DIALOG);
+
+        if (fileChooser.showSaveDialog(null) == JFileChooser.APPROVE_OPTION){
+            /*
+                Remember new start path
```
Comment (swsms) [inline src/editor/files/FileChooser.java:73]: A lot of comments is not always good. If you need a comment inside your code it means the code is not very self-descriptive. In general, comments are good:
- to describe an API like (JavaDoc comments) /**...  */ for classes and methods
- comments within the code only for hard or unobvious places.

If the code is clear enough it doesn't need comments. So, one solution is decomposed your methods into smaller one with human-readable names. Then people could read your code like a book without extra comments.

File: src/editor/files/FileLoader.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public static void saveFile(String fileName, String text, boolean usePrintLn) {
+        /*
+            If file name is empty
+            user should select it
+        */
+        if (fileName == null) {
+            fileName = FileChooser.save();
+        }
+        /*
+          * if user selected nothing
```
Comment (swsms) [inline src/editor/files/FileLoader.java:54]: It is obvious here according to the if statement.

File: src/editor/search/TextSearch.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    /**
+     *  List of indexes of matches inside of text
+     */
+    static  ArrayList<Integer> startIndexes;
+    static  ArrayList<Integer> endIndexes;
+    /**
+     *  Current position inside of matches indexes
+     */
+    static int matchNum;
```
Comment (swsms) [inline src/editor/search/TextSearch.java:24]: There are some concurrency issues. Several threads work with this variable without any kind of synchronization.
If one thread writes a value to a variable, another thread sometimes cannot see the changes. So, some kind of synchronization is needed here. The same with lists.

I would prefer to create an instance TextSearch searcher and pass it to threads which should work with it. To synchronize access to the data you may use the synchronized keyword and special thread-safe collections from java sdk.

**Review by swsms — COMMENTED** [review]
In general, the program looks good, but it should be more thread-safe. Better git's branch naming including "-" instead of camelCase. Like code-simplify.

File: src/editor/search/TextSearch.java (hunk 3 above)
Comment (v-shikhanov) [reply src/editor/search/TextSearch.java:24]: I've tried to realise thread safety  in "search" method, where new threads not start until previous not finished. I'll read about thread-safety collections and will use synchronized keyword. Thanks for attention!

**Pull request closed without merging**